### PR TITLE
Add integrity check before file extraction

### PR DESCRIPTION
**Pull request by Meivyn**
This adds an integrity check before extracting the archives downloaded from BeatMods. If the integrity check isn't passed, then we don't extract anything and just redownload the mod again. Tell me if you see any issues.

### CODE REVIEW

File: ModAssistant/Pages/Mods.xaml.cs [hunk 1]
```diff
@@ -390,22 +393,44 @@ private async Task InstallMod(Mod mod, string directory)
...
-                    if (!string.IsNullOrEmpty(file.Name))
-                    {
-                        await ExtractFile(file, Path.Combine(directory, file.FullName), 3.0, mod.name, 10);
+                        if (!string.IsNullOrEmpty(file.Name))
+                        {
+                            foreach (Mod.DownloadLink download in mod.downloads)
+                            {
+                                foreach (Mod.FileHashes fileHash in download.hashMd5)
+                                {
+                                    if (fileHash.hash == Utils.CalculateMD5FromStream(file.Open()))
```
Comment (lolPants) [inline ModAssistant/Pages/Mods.xaml.cs:419]: Shouldn't `file.Open()` be wrapped with a `using`?

Comment (Meivyn) [reply ModAssistant/Pages/Mods.xaml.cs:419]: Right, I forgot that.

Comment (Meivyn) [reply ModAssistant/Pages/Mods.xaml.cs:419]: Can you fix that? I have to leave now, I'll come a bit later.

**Review by lolPants — CHANGES_REQUESTED** [review]
Doing a quick test, the way this code is currently laid out, the `ZipArchive`s (and by extension all of the entries) are disposed before the code that extracts files runs, resulting in an exception.

**Comment by Meivyn** [comment]
I really don't understand how I didn't see it earlier... I may have made a change that I didn't properly test afterward.

File: ModAssistant/Pages/Mods.xaml.cs (hunk 1 above)
Comment (Meivyn) [reply ModAssistant/Pages/Mods.xaml.cs]: Nevermind, done in the next commit.

**Review by lolPants — APPROVED** [review]
This appears to work. Ideally this would flag up a warning if any files in a mod fail the integrity check (and thus skip the entire mod to prevent weirdness), but as a first pass this is fine.

**Pull request merged**